envoy/server/factory_context.h

```diff
@@ -195,40 +195,29 @@ class ServerFactoryContext : public virtual CommonFactoryContext {
 };
 
 /**
- * Factory context for access loggers that need access to listener properties.
- * This context is supplied to the access log factory when called with the listener context
- * available, such as from downstream HTTP filters.
- * NOTE: this interface is used in proprietary access loggers, please do not delete
- * without reaching to Envoy maintainers first.
+ * Context passed to network and HTTP filters to access server resources.
+ * TODO(mattklein123): When we lock down visibility of the rest of the code, filters should only
+ * access the rest of the server via interfaces exposed here.
  */
-class ListenerAccessLogFactoryContext : public virtual CommonFactoryContext {
+class FactoryContext : public virtual CommonFactoryContext {
 public:
+  ~FactoryContext() override = default;
+
   /**
    * @return Stats::Scope& the listener's stats scope.
    */
   virtual Stats::Scope& listenerScope() PURE;
 
-  /**
-   * @return const envoy::config::core::v3::Metadata& the config metadata associated with this
-   * listener.
-   */
-  virtual const envoy::config::core::v3::Metadata& listenerMetadata() const PURE;
-
   /**
    * @return ProcessContextOptRef an optional reference to the
    * process context. Will be unset when running in validation mode.
    */
   virtual ProcessContextOptRef processContext() PURE;
-};
 
-/**
- * Context passed to network and HTTP filters to access server resources.
- * TODO(mattklein123): When we lock down visibility of the rest of the code, filters should only
- * access the rest of the server via interfaces exposed here.
- */
-class FactoryContext : public virtual ListenerAccessLogFactoryContext {
-public:
-  ~FactoryContext() override = default;
+  /**
+   * @return ListenerInfo description of the listener.
+   */
+  virtual const Network::ListenerInfo& listenerInfo() const PURE;
```

Comment:
I like the consolidation but  it feels like a regression to add more listener info into the non-listener-named factory context.  I suspect it'd be a complete PITA to move this accessor down to the ListenerFactoryContext but have we tried to see how bad it would be?